orderer/common/multichannel/registrar_test.go

```diff
@@ -563,4 +566,144 @@ func TestRegistrar_JoinChannel(t *testing.T) {
 		assert.EqualError(t, err, "channel already exists")
 		assert.Equal(t, types.ChannelInfo{}, info)
 	})
+
+	t.Run("Reject join when join block not etcdraft consensus type", func(t *testing.T) {
+		tmpdir, err := ioutil.TempDir("", "registrar_test-")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpdir)
+
+		ledgerFactory, _ := newLedgerAndFactory(tmpdir, "", nil)
+		mockConsenters := map[string]consensus.Consenter{confSysSolo.Orderer.OrdererType: &mockConsenter{}}
+		config := localconfig.TopLevel{}
+		config.General.BootstrapMethod = "none"
+		config.General.GenesisFile = ""
+		registrar := NewRegistrar(config, ledgerFactory, mockCrypto(), &disabled.Provider{}, cryptoProvider)
+		registrar.Initialize(mockConsenters)
+
+		ledger, err := ledgerFactory.GetOrCreate("my-channel")
+		assert.NoError(t, err)
+		ledger.Append(genesisBlockAppSolo)
+
+		info, err := registrar.JoinChannel("sys-channel", genesisBlockSysSolo)
+
+		assert.EqualError(t, err, "join block consensus type is not configured to etcdraft, found type solo")
+		assert.Equal(t, types.ChannelInfo{}, info)
+	})
+
+	t.Run("Reject join when channel id does not match join block's channelID", func(t *testing.T) {
+		tmpdir, err := ioutil.TempDir("", "registrar_test-")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpdir)
+
+		ledgerFactory, _ := newLedgerAndFactory(tmpdir, "", nil)
+		mockConsenters := map[string]consensus.Consenter{confSysSolo.Orderer.OrdererType: &mockConsenter{}}
+		config := localconfig.TopLevel{}
+		config.General.BootstrapMethod = "none"
+		config.General.GenesisFile = ""
+		registrar := NewRegistrar(config, ledgerFactory, mockCrypto(), &disabled.Provider{}, cryptoProvider)
+		registrar.Initialize(mockConsenters)
+
+		ledger, err := ledgerFactory.GetOrCreate("my-channel")
+		assert.NoError(t, err)
+		ledger.Append(genesisBlockAppSolo)
+
+		info, err := registrar.JoinChannel("wrong-channel-id", genesisBlockSysSolo)
+
+		assert.EqualError(t, err, "config block channelID sys-channel does not match passed channelID wrong-channel-id")
+		assert.Equal(t, types.ChannelInfo{}, info)
+	})
+
+	t.Run("Reject join when join block has both application group and consortiums", func(t *testing.T) {
```

Comment:
In light of the previous comment, this test may need to be tuned.

Reply:
Done